ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
+>
+> `ApplilcationController` inherits from the `VContainer`'s `LifetimeScope` - a class that serves as a dependency injection scope and bootstrapper, where we can bind dependencies.
 
-This approach works, but requires some care: 
- - If you have server and clients of a shared base class, you need to remember that the shared code will run twice on the host. 
- - You also need to take care about code executing in `Start` and `Awake`: if this code runs contemporaneously with the `NetworkManager`'s initialization, it may not know yet whether the player is a host or client.
- - We judged this extra complexity worth it, as it provides a clear road-map to supporting true dedicated servers. 
- - Client-server separation also allows not having god-classes where both client and server code are intermingled. This way, when reading server code, you do not have to mentally skip client code and vice versa. This helps making bigger classes more readable and maintainable. Please note that this pattern can be applied on a case by case basis. If your class never grows too big, having a single `NetworkBehaviour` is perfectly fine.
+## Game state and scene flow
 
-## Connection flow
-The Boss Room network connection flow is owned by the `GameNetPortal`:
- - The Host will invoke either `GameNetPortal.StartHost` or `StartUnityRelayHost` if Unity Relay is being used.
- - The client will invoke either `ClientGameNetPortal.StartClient` or `StartClientUnityRelayModeAsync`.
- - Boss Room's own connection validation logic is performed in `ServerGameNetPortal.ApprovalCheck`, which is plugged in to the `NetworkManager`'s connection approval callback. Here, some basic information about the connection is recorded (including a GUID, to facilitate future reconnect logic), and success or failure is returned. In the future, additional game-level failures will be detected and returned (such as a `ServerFull` scenario). 
+After the initial bootstrap logic is complete, the ApplicationController loads the `MainMenu` scene.
 
-## Data model
-Game data in Boss Room is defined in `ScriptableObjects`. The `ScriptableObjects` are organized by enum and made available in a singleton class: the `GameDataSource`, in particular `ActionDescription` and `CharacterData`. `Actions` represent discrete verbs (like swinging a weapon, or reviving someone), and are substantially data driven. Characters represent both the different player classes, and also monsters, and represent basic details like health, as well as what "Skill" Actions are available to each Character.
+Each scene has its own entrypoint component sitting on a root-level game object. It serves as a scene-specific entrypoint, which, similar to ApplicationController binds dependencies (but these dependencies are local to the specific scene and will be released when the scene unloads). These `State` classes inherit from `LifetimeScope` too, and in the inspector they are set up to have ApplicationController to be their parent scope.
```

Comment:
Your vcontainer section at the bottom is leaking here :P
I'd try not to spread this everywhere. Users reading this for the first time would be interested at first for why do we have a Startup scene in the first place --> singleton creation like NetworkManager and Application controller.
For Game state and scene flow, first thing I'd mention is we have a scene based state machine that manages our overall gameflow.